src/router.ts

```diff
@@ -1,31 +1,63 @@
+import { Ref, ref, watch } from "vue";
 import { createRouter, createWebHistory } from "vue-router";
+import i18n from "./plugins/i18n";
+
+declare module "vue-router" {
+  interface RouteMeta {
+    toolbarTitle: Ref<string>;
+  }
+}
+
+const { locale, t } = i18n.global;
+const rT = (translate: () => string) => {
```

Comment:
what is this for? I don't understand why it's needed. And maybe rT is not a good name then

Reply:
This makes the translation, which is passed as part of the route, reactive. Its naming is derived from the translation functions (called `t` or `tn`) as `reactiveTranslation`.

Without this addition, a title in the toolbar won't change when switching translation.

Reply:
would you mind calling the variable reactiveTranslation? Or autoUpdatedTranslation?
t() is fine since you can look at the import and understand it's for translation. Also translation is used everywhere so it's nice to have a very short name. But rT is confusing since it's hard to understand what's the difference to t() and when to use one over the other. Also it's a local function that's only used in this file, so no reason to give it an understandable name.

Reply:
Also added a static type definition to the function, which makes the return type and the function definition explicit and independent of the implementation.